mobify/schemer
# [WIP] Initial commit for Schemer

Status: **Opened for visibility**
Reviewers: @tedtate @Helen-Mobify @jnurse @mobify-derrick @ellenmobify 
## IMPORTANT NOTE

Schemer is under heavy development (read: shit will constantly change/break) at this stage!
## Jira Tickets:
- [x] [CSOPS-1340](https://mobify.atlassian.net/browse/CSOPS-1340)
- [x] [CSOPS-1339](https://mobify.atlassian.net/browse/CSOPS-1339)
- [x] [CSOPS-1337](https://mobify.atlassian.net/browse/CSOPS-1337)
- [x] [CSOPS-1338](https://mobify.atlassian.net/browse/CSOPS-1338)
- [x] [CSOPS-1429](https://mobify.atlassian.net/browse/CSOPS-1429)
- [x] [CSOPS-1405](https://mobify.atlassian.net/browse/CSOPS-1405)
## Todos:
- [x] Engineer +2
- [x] Scripting support (run unsupervised)
- [x] Test integration with sample CircleCI
- [x] MobifyJS support?
- [x] Add Ignore support. Usecase: A value that changes every time, even on the same fixture. Eg: cache-busted string.
- [x] Embed view and fixture path in the schema
### Feedback:
- [x] Remove bower_components from repo and add to gitignore, update readme to add `bower install`
## How to Test
- Switch to an Adaptive 1.0 project with integration tests written (ensures testing infrastructure is set up)
- npm install schemer from this repo
- Run Schemer in interactive mode (node_modules/schemer/index.js --interactive)
- Navigate to localhost:3000/schemer
- Ensure that views from the selected project show up in the list. All schema should match
- Modify a view in the project to change generated schema. Both interactive and unsupervised mode should show fail.
